### PR TITLE
DM-2961: Port psfex support from HSC

### PR DESCRIPTION
**Pull request by mjuric**

### CODE REVIEW

File: ups/eupspkg.cfg.sh [hunk 1]
```diff
@@ -1,4 +1,18 @@
 # EupsPkg config file. Sourced by 'eupspkg'
 
-CONFIGURE_OPTIONS="--prefix $PREFIX --disable-fortran --enable-shared"
 MAKE_INSTALL_TARGETS="-j1 install"
+
+config(){
+	clean_old_install
```
Comment (mjuric) [inline ups/eupspkg.cfg.sh]: This usually goes into `install()` (see https://github.com/lsst/boost/blob/master/ups/eupspkg.cfg.sh)

File: ups/eupspkg.cfg.sh [hunk 2]
```diff
@@ -1,4 +1,18 @@
...
+}
+
+build() {
+	./configure --prefix $PREFIX --disable-fortran --enable-shared $1
+	make && make -j1 install
+}
+
+install()
+{
+	build --enable-single
```
Comment (mjuric) [inline ups/eupspkg.cfg.sh]: Watch out -- you're call the build function above again (in effect, rebuilding the package).

**Comment by mjuric** [comment]
A general question about the changes to `eupspkg.cfg.sh` -- why did you need to override `config`, `build`, and `install`? I'd expect that setting `CONFIGURE_OPTIONS` and maybe `MAKE_INSTALL_FLAGS` would get the job done?

**Comment by mjuric** [comment]
Is it that multi-job install on fftw has a bug (so you want to pass '-j1' to make?)

**Comment by PaulPrice** [comment]
We need to configure, build and install fftw twice --- once for double and once for float, to produce the fftw and fftwf libraries.

**Comment by mjuric** [comment]
Oy vey! That at least demands a 10-line comment block on the top of the file to explain what's going on :).

**Comment by PaulPrice** [comment]
Should we, perhaps, implement this as a separate external package?

**Comment by mjuric** [comment]
More on topic: I'd then propose to do the following instead:
- in prep, call `default_prep` to expand+patch the tarball, and then copy the result into two directories (e.g., sp/ and dp/).
- then override `config`, `build`, and `install` to just `cd` into those directories, do the configurations (as appropriate) and builds and the installs.

I think that would be cleaner and more self-documenting.

File: ups/eupspkg.cfg.sh (hunk 1 above)
Comment (natelust) [inline ups/eupspkg.cfg.sh]: Changed

**Comment by natelust** [comment]
I have adopted your recommendation with with two directory strategy. The source for single and double precision code can be found under source/sp and source/dp respectively. Each still installs to the same path

**Pull request merged**